src/target/adi.c

```diff
@@ -699,5 +837,21 @@ void adi_ap_component_probe(
 				break;
 			}
 		}
+
+		/* Start out by making a fake AP to use for all the reads */
+		adiv5_access_port_s base_ap = {
+			.dp = ap->dp,
+			.base = base_address,
+		};
```

Comment:
You should be able to skip this (it is used in ADIv6 to simulate an AP for the debug resource bus accesses that that parser has to make) and just pass `ap` to the parse function below.